### ISSUE TITLE
Support disabling consul

### ISSUE THREAD

**Issue by geek** [joyent/containerpilot#551]
Allow ContainerPilot to execute and startup jobs, even jobs that register ports and have health checks, without consul. If the `consul` configuration is set to `false` then consul is being disabled. When this occurs the jobs should still be initialized but there shouldn't be attempts to register the jobs/services with consul. Similarly, there shouldn't be support for consul watches. If watches are in use with `consul: false` then there should be an error. There are likely other edge cases too, but I do think this is possible and a valid use case for ContainerPilot to act as an init process without requiring consul.

**Comment by riot-jayb** [comment]
Agreed. I submitted a PR for this back in the 2.X days and was told that in 3.X there would be an option for this. Coming back nearly 2 years later, it still seems that consul is tightly coupled with containerpilot.

Can the maintainers consider a yay/nay statement on this proposal quickly? I think there would be many within the community who would be happy to contribute this update. As proposed by @geek, this should not be a large change or burdensome for maintenance.

**Comment by riot-jayb** [comment]
@cheapRoc @tgross would you accept some external contributions here?

**Comment by ahelal** [comment]
would be very interesting if this is implemented or open for PR

**Comment by gbmeuk** [comment]
does running [consul as job](https://www.joyent.com/containerpilot/docs/configuration/consul) and [setting the agent in dev mode](https://www.consul.io/docs/agent/options.html#_dev) work for you?
